### DISCUSSION TITLE
How should I store users third-party API keys in Supabase?

### DISCUSSION THREAD

**Discussion by AlexanderCryptoOwl** [supabase/supabase#5840] [Questions]
Hi there :),

Loving Supabse and recently launched  https://cryptoowl.co/ with you guys. I now like to offer a feature to users where they can provide me with their Binance, Coinbase etc. API keys and I will do automated trading with their accounts.

My question is since API keys are very sensitive, is there some best practices in Supabse I should be mindful of?

Some notes/thinking
1. User can add their key but are never able to read it afterwards only change the key (feels safer).
2. I will be using cloud functions to run my Python trading Algo (use Supabase master key).

I know Supabase has the pgcrypto and rpc function trigger but could my cloud function access the value of the protected key?

**Comment by charlesoestergaard** [comment]
Not related too question but you have a z-index error -> 

![chrome_zkH6IDOrsO](https://user-images.githubusercontent.com/59709554/157040380-f331808d-6898-4998-982c-f30632efb908.png)

**Reply by AlexanderCryptoOwl** [reply]
Haha true thanks will fix!

**Reply by charlesoestergaard** [reply]
> 

Mabye you can help me with my problem? -> https://github.com/supabase/supabase/discussions/5843
